File: Visitor/python-source/rooms/Surface.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Surface:
    """Поверхность комнаты: из чего сделана и какого цвета."""

    material: str = "unknown"
    color: str = "unknown"

    def __str__(self) -> str:
        return f"{self.material}, {self.color}"

File: JavaSource/src/main/java/ru/akhcheck/patterns/visitor/rooms/Surface.java
package ru.akhcheck.patterns.visitor.rooms;

/** Поверхность комнаты: из чего сделана и какого цвета. */
public class Surface {
    private String material = "unknown";
    private String color = "unknown";

    public String getMaterial() {
        return material;
    }

    public void setMaterial(String material) {
        this.material = material;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    @Override
    public String toString() {
        return material + ", " + color;
    }
}
